lib/reservation.rb

```diff
@@ -0,0 +1,22 @@
+
+
+module Hotel
+  class Reservation
+    attr_reader :start_date, :end_date, :room_id, :reservation_dates, :total_cost
+
+    def initialize(reservation_dates, room_id)
+
+      @reservation_dates = reservation_dates
```

Comment:
The name `reservation_dates` is a little redundant, since we're already in the `Reservation` class. Something like `dates` or `date_range` might be a better name.